modules/au.com.trgtd.tr.task.messages/test/unit/src/au/com/trgtd/tr/task/messages/MessageParserTest.java

```diff
@@ -0,0 +1,126 @@
+package au.com.trgtd.tr.task.messages;
+
+import au.com.trgtd.tr.task.messages.MessageParser.Message;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Date;
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.List;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+public class MessageParserTest {
+
+    private MessageParser parser = new MessageParser();
+    private LocalDate localDate = LocalDate.of(2014, Month.MARCH, 26);
+
+    final String xml1 = "<msgs>"
+            + "<msg date='2014/03/27' type='A'>message1</msg>"
+            + "<msg date='2014/03/30' type='B'>message2</msg>"
+            + "</msgs>";
+
+    final String xml2 = "<msgs>"
+            + "<msg date='2014/03/27' type='N'>nonMemberMessage</msg>"
+            + "<msg date='2014/03/30' type='M'>memberMessage</msg>"
+            + "</msgs>";
+
+    @Test
+    public void parsingValidButEmptyXml_returnsNoMessages() throws Exception {
+        assertTrue(parse("<msgs/>", localDate).isEmpty());
+    }
+
+    @Test
+    public void parsingXml1_withDateBeforeBoth_returnsBothMessages() throws Exception {
+        List<Message> messages = parse(xml1, localDate);
+        assertTrue(messages.size() == 2);
+        assertMessage(messages.get(0), LocalDate.of(2014, Month.MARCH, 27), "a", "message1");
+        assertMessage(messages.get(1), LocalDate.of(2014, Month.MARCH, 30), "b", "message2");
+    }
+
+    @Test
+    public void parsingXml1_withDateAfterFirstButBeforeSecond_returnsNewMessage() throws Exception {
+        List<Message> messages = parse(xml1, localDate.plusDays(2));
+        assertTrue(messages.size() == 1);
+        assertMessage(messages.get(0), LocalDate.of(2014, Month.MARCH, 30), "b", "message2");
+    }
+
+    @Test
+    public void parsingXml2asMember_returnsMemberMsg() throws Exception {
+        Boolean isMember = true;
+        List<Message> messages = parse(xml2, localDate, isMember);
+        assertTrue(messages.size() == 1);
+        assertMessage(messages.get(0), LocalDate.of(2014, Month.MARCH, 30), "m", "memberMessage");
+    }
+
+    @Test
+    public void parsingXml2asNonMember_returnsNonMemberMsg() throws Exception {
+        Boolean isMember = false;
+        List<Message> messages = parse(xml2, localDate, isMember);
+        assertTrue(messages.size() == 1);
+        assertMessage(messages.get(0), LocalDate.of(2014, Month.MARCH, 27), "n", "nonMemberMessage");
+    }
+
+    @Test
+    public void parseFile() throws Exception {
+        Path path = Paths.get(getClass().getResource("messages.xml").getPath());
+        String xml = new String(Files.readAllBytes(path));
+        List<Message> messages = parse(xml, localDate);
+        assertTrue(messages.size() == 1);
+        assertMessage(messages.get(0), LocalDate.of(2014, Month.MARCH, 27), "a", null);
+        String text = messages.get(0).text;
+        assertEquals(431, text.length());
```

Comment:
431? What does it mean? Why 431? Probably the length of the original message?
I would prefer variable like
const int LENGHT_OF_SAMPLE_MESSAGE  = 431;